src/Datastore/Key.php

```diff
@@ -348,6 +348,60 @@ public function pathEnd()
     }
 
     /**
+     * Get the last pathElement identifier.
+     *
+     * If the key is incomplete, returns `null`.
+     *
+     * Example:
+     * ```
+     * $lastPathElementIndentifier = $key->pathEndIdentifier();
+     * ```
+     *
+     * @return string|null
+     */
+    public function pathEndIdentifier()
+    {
+        $end = $this->pathEnd();
+
+        if (isset($end['id'])) {
+            return $end['id'];
```

Comment:
Is $end['id'] an integer, or string? If it's integer, you may need to change the docblock?